cmd/datasetIngestor/main_test.go

```diff
@@ -11,6 +12,8 @@ import (
 // It captures the stdout, runs the main function, and checks if the output contains the expected strings.
 // This just checks if the main function prints the help message.
 func TestMainOutput(t *testing.T) {
+	flag.CommandLine = flag.NewFlagSet("main output", flag.ExitOnError)
```

Comment:
not too sure I understand why this is needed

Reply:
Because the flag library is persistent, and if we call the main() function multiple times per test, and we're using the flags library for defining flags in the main function, then we'll end up redefining the same flags which will cause a panic during testing only. 

My second set of commits introduced this fix, which effectively hides/deletes the old flag definitions for each test.

Reply:
ah ok, thanks. Should we keep it for the last commit as well then?

Reply:
Nope, because for some reason, that caused another issue with this test, and the simplest solution I could think of is to just remove it. The test wasn't written by me btw. so I haven't looked much into it why it doesn't like to start with an empty list of flags.